src/app.ts

```diff
@@ -179,50 +179,63 @@ export async function newApp(
   registerAPIRoutes(app, APIBasePath, allowNotifyIPSourceRange);
   registerPagoPARoutes(app, PagoPABasePath, allowPagoPAIPSourceRange);
 
-  const timer = idpMetadataUpdater(
+  const refreshTimeMilliseconds =
+    container.resolve<number>(IDP_REFRESH_INTERVAL_SECONDS) * 1000;
+  const idpMetadataUpdateTimer = idpMetadataUpdater(
     app,
-    container.resolve<number>(IDP_REFRESH_INTERVAL_SECONDS) * 1000
+    refreshTimeMilliseconds
   );
   app.on("server:stop", () => {
-    clearInterval(timer);
+    clearInterval(idpMetadataUpdateTimer);
   });
   return app;
 }
 
+/**
+ * Initializes SpidStrategy for passport and setup /login route.
+ */
 async function registerLoginRoute(app: Express): Promise<void> {
   // Add the strategy to authenticate the proxy to SPID.
-  passport.use(
-    "spid",
-    await container.resolve<Promise<passport.Strategy>>(SPID_STRATEGY)
+  const spidStrategy = await container.resolve<Promise<passport.Strategy>>(
+    SPID_STRATEGY
   );
+  passport.use("spid", spidStrategy);
   const spidAuth = passport.authenticate("spid", { session: false });
   app.get("/login", spidAuth);
 }
 
+/**
+ * Clears SPID_STRATEGY cache from awilix.
+ * Then /login route will be overridden with an updated SPID_STRATEGY.
+ */
 async function clearAndReloadSpidStrategy(
   app: Express,
   onRefresh?: () => void
 ): Promise<void> {
   log.info("Started Spid strategy re-initialization ...");
 
+  // Clear cache for SPID_STRATEGY definition (singleton lifetime) on awilix.
+  // The next resolution of SPID_STRATEGY will reinstantiate the registered class.
   container.cache.delete(SPID_STRATEGY);
 
   passport.unuse("spid");
 
-  // Remove /login route from Express router stack
-  // tslint:disable: no-any
-  // tslint:disable-next-line: no-object-mutation
-  app._router.stack = app._router.stack.filter((_: any) => {
+  // tslint:disable-next-line: no-any
+  const isLoginRoute = (route: any) => {
     if (
```

Comment:
`if` is not needed: when `x` is a boolean expression, like in this case, `if(x) { return true; } else { return false; }` is the same as just `x`

Reply:
so:

```ts
const isLoginRoute = (route: any) => route.route &&
      route.route.path === "/login" &&
      route.route.methods &&
      route.route.methods.get;
```

Reply:
Fixed in a6e4c2a